### ISSUE TITLE
Pretraining Piccolo2 Model from Scratch

### ISSUE THREAD

**Issue by binhna** [hjq133/piccolo-embedding#11]
Hi,

Thank you for your great work.

I noticed that you set `piccolo2-large-zh-0417` as your pretrained model in the `ft.sh` file, which I assume stands for `finetune.sh`. My question is:

1. Is this repository only for fine-tuning the `Piccolo2` model? 
2. If so, do you have the code to pretrain the `Piccolo2` model from scratch instead of just fine-tuning it?

Thank you.

**Comment by hjq133** [comment]
hello，as we mentioned in hf: https://huggingface.co/sensenova/piccolo-large-zh-v2, we use [stella-v3.5](https://huggingface.co/dunzhang/stella-mrl-large-zh-v3.5-1792d) as our pretrained model, and finetune it using multi task hybrid loss training. You can also use this repo for your pretraining, because most of the pretraining data is labeled as 'Retri' data：containing 3 columns: text, text_pos, text_neg, and we implement standard InfoNCE with in-batch-negative in this repo.

**Comment by hjq133** [comment]
In general embedding training, the loss used for pre-training and fine-tuning is almost the same: infonce loss. The difference lies in the different datasets used and the sources of negative samples. For pre-training, negative samples mostly come from in-batch negatives, while for fine-tuning, negative samples are hard negatives obtained by mining.
